### PR TITLE
options: allow passing in TLS credentials

### LINKED ISSUE #44: options: provide option to use TLS configurations
Coming here from https://github.com/census-instrumentation/opencensus-service/issues/367
for parity we need this exporter to be able to start with TLS client credentials. For the OpenCensus receiver aka server side, I've mailed out https://github.com/census-instrumentation/opencensus-service/pull/415.

### PR DESCRIPTION
**Pull request by odeke-em**
Allow passing in TLS credentials.
This PR is the sister to ocagent's
https://github.com/census-instrumentation/opencensus-service/pull/415

We shall have the ability to have a secure ocagent and
applications that share the certificate through whatever
certificate sharing mechanism e.g. rotation, file etc.

For an example
```go
func main() {
	// Please take at look at https://godoc.org/google.golang.org/grpc/credentials#TransportCredentials
	// for ways on how to initialize gRPC TransportCredentials.
	creds, err := credentials.NewClientTLSFromFile("my-cert.pem", "")
	if err != nil {
		log.Fatalf("Failed to create gRPC client TLS credentials: %v", err)
	}

	exp, err := ocagent.NewExporter(ocagent.WithTLSCredentials(creds), ocagent.WithServiceName("engine"))
	if err != nil {
		log.Fatalf("Failed to create the agent exporter: %v", err)
	}
	defer exp.Stop()

	// Now register it as a trace exporter.
	trace.RegisterExporter(exp)

	// Then use the OpenCensus tracing library, like we normally would.
	ctx, span := trace.StartSpan(context.Background(), "Securely-Talking-To-Agent-Span")
	defer span.End()

	for i := 0; i < 10; i++ {
		_, iSpan := trace.StartSpan(ctx, fmt.Sprintf("Sample-%d", i))
		<-time.After(6 * time.Millisecond)
		iSpan.End()
	}
}
```

Fixes #44

### CODE REVIEW

File: options.go [hunk 1]
```diff
@@ -103,3 +107,22 @@ func (h headerSetter) withExporter(e *Exporter) {
...
+
+var _ ExporterOption = (*clientCredentials)(nil)
+
+// WithTLSCredentials allows the connection to use TLS credentials
+// when talking to the server. It takes in grpc.TransportCredentials instead
+// of say a Certificate file or a tls.Certificate, because the retrieving
+// these credentials can be done in many ways e.g. plain file, in code tls.Config
+// or by certificate rotation, so it is up to the caller to decide what to use.
+func WithTLSCredentials(creds credentials.TransportCredentials) ExporterOption {
+	return &clientCredentials{TransportCredentials: creds}
```
Comment (songy23) [inline options.go]: nit: consider adding a check that `WithInsecure` and `WithTLSCredentials` shouldn't be called at the same time.

Comment (odeke-em) [reply options.go]: Oh, if you see in New, I do that
https://github.com/census-ecosystem/opencensus-go-exporter-ocagent/blob/5a6e73f487e155ca74647bf2acdcf88e09133e2a/ocagent.go#L263-L267

Comment (odeke-em) [reply options.go]: so they are mutually exclusive!

**Comment by odeke-em** [comment]
Thank you for the reviews @songy23 and @sjkaris!

**Pull request merged**